src/Symfony/Component/Serializer/Normalizer/AbstractObjectNormalizer.php

```diff
@@ -38,11 +41,21 @@
     private $attributesCache = array();
     private $cache = array();
 
-    public function __construct(ClassMetadataFactoryInterface $classMetadataFactory = null, NameConverterInterface $nameConverter = null, PropertyTypeExtractorInterface $propertyTypeExtractor = null)
+    /**
+     * @var ClassDiscriminatorResolverInterface|null
+     */
+    protected $classDiscriminatorResolver;
+
+    public function __construct(ClassMetadataFactoryInterface $classMetadataFactory = null, NameConverterInterface $nameConverter = null, PropertyTypeExtractorInterface $propertyTypeExtractor = null, ClassDiscriminatorResolverInterface $classDiscriminatorResolver = null)
```

Comment:
Do we really want to keep exposing this `$classDiscriminatorResolver` argument? Configuring everything through metadata looks enough to me, isn't it?

Reply:
I would keep it. We may leverage this extension point with API Platform to remove a lot of code related to `abstract` class handling. But we need to hook our own metadata system.